app/models/piece.rb

```diff
@@ -32,4 +32,24 @@ def set_image
     color ? color_string = "white" : color_string = "black"
     self.image_name ||= "#{color_string}-#{piece_type.downcase}.png"
   end
+
+  def move_to!(new_x, new_y)
```

Comment:
@FallGambit/chess-team From several of the method descriptions in the storyboard I think the way Jeff describes calling valid_move? from inside any move_to! action is the way it was designed/intended to work.